lib/ps.js

```diff
@@ -1,131 +1,103 @@
-      break
-    case 'aix':
-      headers = 'pcpu,rsssize,pid,etime'
-      break
-    default:
-      headers = 'pcpu,rss,pid,etimes'
-      doParseTime = false
+  if (i >= 0) { // Seconds
+    time += parseInt(tpart[i--], 10) * 1000
   }
+  if (i >= 0) { // Minutes
+    time += parseInt(tpart[i--], 10) * 60000
+  }
+  if (i >= 0) { // Hours
+    time += parseInt(tpart[i--], 10) * 3600000
+  }
+  if (i >= 0) { // Days
+    time += parseInt(tpart[i--], 10) * 86400000
+  }
+  return time
+}
 
-  var args = ['-o', headers, '-p', pids]
-
-  var ps = spawn('ps', args)
-  var error
-  var stdout
-  var stderr = ''
-
-  ps.on('error', function (err) {
-    error = err
-  })
+/**
+  * Get pid informations through ps command.
+  * @param  {Number[]} pids
+  * @param  {Object} options
+  * @param  {Function} done(err, stat)
+  */
+function ps (pids, options, done) {
+  var pArg = pids.join(',')
 
-  ps.stdout.on('data', function (d) {
-    stdout += d.toString()
-  })
+  var args = ['-o', 'etime,pid,ppid,pcpu,rss,time', '-p', pArg]
 
-  ps.stderr.on('data', function (d) {
-    stderr += d.toString()
-  })
+  if (PLATFORM === 'aix') {
+    args = ['-o', 'etime,pid,ppid,pcpu,rsssize,time', '-p', pArg]
+  }
 
-  ps.on('close', function (code) {
+  bin('ps', args, function (err, stdout) {
+    if (err) return done(err)
     var date = Date.now()
-    if (error) return done(error, null)
-    if (code !== 0) return done(new Error('ps exited with code ' + code + ', command was "ps ' + args.join(' ') + '" and error: ' + stderr))
-    if (!stdout) {
-      return done(null, pids.map(function (pid) {
-        return {cpu: 0, memory: 0, pid: parseInt(pid, 10), time: 0, start: new Date(date)}
-      }))
-    }
+
+    // Example of stdout on *nix.
+    // ELAPSED: format is [[dd-]hh:]mm:ss
+    // RSS: is counted as blocks of 1024 bytes
+    // TIME: format is [[dd-]hh:]mm:ss
+    //
+    //    ELAPSED   PID  PPID  %CPU     RSS        TIME
+    // 2-40:50:53   430     1   3.0    5145  1-02:03:04
+    //   40:50:53   432   430   0.0    2364  1-01:02:03
+    //   01:50:50   727     1  10.0  348932       14:27
+    //      00:20  7166     1   0.1    3756        0:00
+
+    // Example of stdout on Darwin
+    // ELAPSED: format is [[dd-]hh:]mm:ss
+    // RSS: is counted as blocks of 1024 bytes
+    // TIME: format is [[dd-]hh:]ss:mm.pp (pp is the percentage of a minute)
+    //
+    //    ELAPSED   PID  PPID  %CPU     RSS           TIME
+    // 2-40:50:53   430     1   3.0    5145  1-02:03:04.07
+    //   40:50:53   432   430   0.0    2364  1-01:02:03.10
+    //   01:50:50   727     1  10.0  348932       14:27.26
+    //      00:20  7166     1   0.1    3756        0:00.02
 
     stdout = stdout.split(os.EOL)
 
     var statistics = {}
     for (var i = 1; i < stdout.length; i++) {
-      stdout[i] = stdout[i].trim()
+      var line = stdout[i].trim().replace(/\s\s+/g, ' ').split(' ')
 
-      if (!stdout[i]) {
+      if (!line || line.length !== 6) {
```

Comment:
If I understand this, you'll skip th eline if it hasn't all the data? How can this happend?

Reply:
It cannot, but who knows 🙈

Reply:
Then remove this part from the condition :laughing: